src/python/deepgnn/graph_engine/snark/convert.py

```diff
@@ -29,7 +30,7 @@ def __init__(
         graph_path: str,
         meta_path: str,
         output_dir: str,
-        decoder_type: DecoderType = DecoderType.JSON,
+        decoder: Optional[DecoderType] = None,
```

Comment:
why a new default value is `None`? I thought `PipeDispatcher` doesn't initialize it and there will be an exception. It seems this parameter can't be optional.

Reply:
TypeVar does not work with default arguments. The dispatchers just pass this value to the convert_process/output function which is where the "if decoder_type is None: decoder_type = JsonDecoder()".

Reply:
TypeVar is the only way to handle inheritance with python typing. The optional is required to support the default value with a TypeVar.

Reply:
Default value now only for MultiWorkerConverter not dispatchers.